libraries.cmake/hdf5.cmake

```diff
@@ -0,0 +1,90 @@
+##################################################
+###       HDF5                                 ###
+##################################################
+MACRO( OPENMS_CONTRIB_BUILD_HDF5 )
+
+  # Minimal cmake requirement for building HDF5 1.10.5
+  cmake_minimum_required(VERSION 3.10.0 FATAL_ERROR)
```

Comment:
this may be a slight hickup, @jpfeuffer will this cause any problems?

Reply:
Hmm not necessarily. I am okay with upgrading to newer CMake's (I don't think all our build machine fulfill this requirement right now).